### ISSUE TITLE
It should be possible to access inherited nested types through an unbound generic type in typeof expression

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#9850]
This is port of http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/542581.
It was resolved as "No Repro", but the unit-test (Microsoft.CodeAnalysis.VisualBasic.UnitTests.CodeGenGetTypeOperator.CodeGen_GetType_InheritedNestedTypeThroughUnboundGeneric) remained skipped. It looks like the behavior got regressed.

--------------------
```cs
using System;
 
class C
{
    public class E { }
 
    static void Main()
    {
        Console.WriteLine(typeof(D<>.E));
    }
}
 
class D<T> : C { }
```

**ACTUAL RESULT**: error CS0426: The type name 'E' does not exist in the type 'D<<Error>>'

**EXPECTED RESULT**: no errors, prints C+E
 
See test `Roslyn.Compilers.CSharp.UnitTests.CodeGen.CodeGenTests.TestTypeofInheritedNestedTypeThroughUnboundGeneric`

**Comment by gafter** [comment]
This test is no longer skipped, but instead tests that the compiler remains broken!
